applications/StructuralMechanicsApplication/custom_response_functions/adjoint_conditions/adjoint_semi_analytic_point_load_condition.cpp

```diff
@@ -0,0 +1,212 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:		 BSD License
+//					 license: structural_mechanics_application/license.txt
+//
+//  Main authors:    Martin Fusseder, https://github.com/MFusseder
+//
+
+// System includes
+
+
+// External includes
+
+
+// Project includes
+#include "adjoint_semi_analytic_point_load_condition.h"
+#include "structural_mechanics_application_variables.h"
+#include "includes/checks.h"
+
+namespace Kratos
+{
+    //******************************* CONSTRUCTOR ****************************************
+    //************************************************************************************
+
+    AdjointSemiAnalyticPointLoadCondition::AdjointSemiAnalyticPointLoadCondition(Condition::Pointer pPrimalCondition )
+        : AdjointSemiAnalyticBaseCondition(pPrimalCondition )
+    {
+    }
+
+    //********************************* CREATE *******************************************
+    //************************************************************************************
+
+    Condition::Pointer AdjointSemiAnalyticPointLoadCondition::Create(Condition::Pointer pPrimalCondition ) const
+    {
+        return Kratos::make_shared<AdjointSemiAnalyticPointLoadCondition>( pPrimalCondition );
+    }
+
+    //******************************* DESTRUCTOR *****************************************
+    //************************************************************************************
+
+    AdjointSemiAnalyticPointLoadCondition::~AdjointSemiAnalyticPointLoadCondition()
+    {
+    }
+
+    //************************************************************************************
+    //************************************************************************************
+
+    void AdjointSemiAnalyticPointLoadCondition::EquationIdVector(EquationIdVectorType& rResult, ProcessInfo& rCurrentProcessInfo )
+    {
+        KRATOS_TRY
+
+        const unsigned int number_of_nodes = GetGeometry().size();
+        const unsigned int dim = GetGeometry().WorkingSpaceDimension();
+        if (rResult.size() != dim * number_of_nodes)
+        {
+            rResult.resize(dim*number_of_nodes,false);
+        }
+
+        const unsigned int pos = this->GetGeometry()[0].GetDofPosition(ADJOINT_DISPLACEMENT_X);
+
+        if(dim == 2)
+        {
+            for (unsigned int i = 0; i < number_of_nodes; ++i)
```

Comment:
Here we usually use `IndexType` => same as for `SizeType`

Reply:
Done